### PR TITLE
Improve ukrainian translation, part 1

### PR DESCRIPTION
**Pull request by sfkulyk**
## Summary
Updates to ukrainian translation (both help file and interface)

 
## Checklist
- [x] I have followed the [contributing guidelines](https://github.com/FarGroup/FarManager/blob/master/CONTRIBUTING.md).
- [x] I have discussed this with project maintainers:  <br/>
If not checked, I accept that this work might be rejected in favor of a different grand plan.<br/>

## Details
None

### CODE REVIEW

**Comment by sfkulyk** [comment]
Updates to  ukrainian help file
Also update ukrainian interface a bit.

File: far/FarUkr.hlf.m4 [hunk 1]
```diff
@@ -252,19 +251,18 @@ user profile folder (#%APPDATA%\\Far Manager\\Profile# by default).
  Location of Far template configuration file (overrides the ini file).
 
  #-title[:<title>]#
- If <title> string is provided, use it as the window title; otherwise
-inherit the console window's title. Macro #%Default# in the custom title
-string will be replaced with the standard context-dependent Far window's
-title.
+ Якщо вказан <title>, він буде використан для заповнення заголовку вікна FAR; якщо
+опція вказана без значенню, заголовок вікна буде взятий з заголовка консолі. Макрос #%Default#
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:255]: "без значення".
It might be better to reword this one as well.

File: far/FarUkr.hlf.m4 [hunk 2]
```diff
@@ -278,13 +276,13 @@ title.
...
+ - ^<wrap>для каталогів та архівів Far покаже їх зміст;
  - ^<wrap>для файлів - відбудеться перехід у каталог із файлом і
     позиціонування на вказаний файл, якщо він існує;
- - ^<wrap>when prefixes specified (simultaneous use with common paths allowed)
-passive command executes first (passive panel activates temporary).
-Single letter prefixes A-Z or conflicted with disk letter will be ignored.
- Example: far arc:c:\\Far20.7z "lua:msgbox('Far Manager','Successfully started')"
+ - ^<wrap>якщо префікс указано (можно використовувати із звичайним path)
+пасивна команда буде виконана перша (пасивна панель буде тимчасово активована).
+Префікси з одної літери A-Z чи конфліктуючи з літерою диска будуть ігноруватись.
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:284]: "конфліктуючі".

File: far/FarUkr.hlf.m4 [hunk 3]
```diff
@@ -313,7 +311,7 @@ $ #Menu control commands#
  Filter menu or list items                          #Ctrl+Alt+F,RAlt#
  Lock filter                                             #Ctrl+Alt+L#
 
- See also the list of ~macro keys~@KeyMacroMenuList@, available in the menus.
+ Також дивиться список ~макро клавиш~@KeyMacroMenuList@, що доступні у меню.
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:314]: "дивіться"

File: far/FarUkr.hlf.m4 [hunk 4]
```diff
@@ -333,14 +331,14 @@ $ #Команди керування панелями#
...
  Змінити ширину                               #Ctrl+Left, Ctrl+Right#
  (при пустому командному рядку)
  Відновити ширину панелей за замовчуванням             #Ctrl+Numpad5#
- Restore default panels height                     #Ctrl+Alt+Numpad5#
+ Відновити висоту панелей за замовчуванням         #Ctrl+Alt+Numpad5#
  Сховати/Показати лінійку функціональних клавіш              #Ctrl+B#
- Toggle total and free size show mode                  #Ctrl+Shift+S#
- in bytes (if possible) or with size suffixes K/M/G/T
+ Переключити показ загального та вільного розміру      #Ctrl+Shift+S#
+ в байтах (якщо можливо) або розмір з суффіксами K/M/G/T
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:341]: "суфіксами"

File: far/FarUkr.hlf.m4 [hunk 5]
```diff
@@ -448,29 +446,29 @@ $ #Команди керування панелями#
 
 @PanelCmdSort
 $ #Sort modes#
- The sort modes menu is called by #Ctrl+F12# and applies to the currently
-active panel. The following sort modes are available:
+ Меню вибору сортування можно викликати натисканням #Ctrl+F12#, після вибору, режим буде застосовано
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:449]: "можна"

File: far/FarUkr.hlf.m4 [hunk 6]
```diff
@@ -448,29 +446,29 @@ $ #Команди керування панелями#
...
+до активної панелі. Можно обрати режим сортування із списка:
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:450]: "можна"
"зі списку"

File: far/FarUkr.hlf.m4 [hunk 7]
```diff
@@ -448,29 +446,29 @@ $ #Команди керування панелями#
...
- The #+# key sets the sorting order to be direct.
- The #-# key sets the sorting order to be reversed.
- The #*# key changes the sorting order to be reversed.
+ #+# встановлює прямий порядок сортування.
+ #-# встановлює зворотній порядок сортування.
+ #*# змінює порядок сортування на протилежний.
 
- Use group sorting                                        #Shift+F11#
- Show selected files first                                #Shift+F12#
+ Використовувати групову сортировку                 #Shift+F11#
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:466]: "групове сортування"

File: far/FarUkr.hlf.m4 [hunk 8]
```diff
@@ -448,29 +446,29 @@ $ #Команди керування панелями#
...
+ Відображати спочатку відмічені файли               #Shift+F12#
 
- You can ~fine-tune~@PanelSortCriteria@ sort modes by pressing #F4#.
+ Можно обрати ~тонке налаштування~@PanelSortCriteria@ режиму сортировки натиснув #F4#.
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:469]: "можна"
"обрати тонке налаштування режиму" sounds complex, maybe "додатково налаштувати режим" or something?

File: far/FarUkr.hlf.m4 [hunk 9]
```diff
@@ -448,29 +446,29 @@ $ #Команди керування панелями#
...
 
- See also: common ~menu~@MenuCmd@ keyboard commands.
+ Також дивиться: Основне ~меню~@MenuCmd@ клавіатурних команд.
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:471]: "дивіться"

File: far/FarUkr.hlf.m4 [hunk 10]
```diff
@@ -518,8 +516,8 @@ $ #Швидкий пошук у панелях#
 переміститься курсор.
 
  За допомогою #Ctrl+Enter# можна циклічно переміщуватися по елементах
-панелі у відповідності до введеної у рядку файлової маски. #Ctrl+Shift+Enter# allows to
-cycle backwards.
+панелі у відповідності до введеної у рядку файлової маски. #Ctrl+Shift+Enter# дозволяє
+переміщуватись в обратному напрямку.
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:520]: зворотньому

File: far/FarUkr.hlf.m4 [hunk 11]
```diff
@@ -696,14 +694,14 @@ Alt+F9 перемикає екран з 25 ліній на 50 і навпаки.
...
  Перейменування або перенесення файла під курсором         #Shift+F6#
- For folders: if the specified path (absolute or relative) points to an
-existing folder, the source folder is moved inside that folder. Otherwise the
-folder is renamed/moved to the new path.
- E.g. when moving #c:\folder1\# to #d:\folder2\#:
- - ^<wrap>if #d:\folder2\# exists, contents of #c:\folder1\# is
-moved into #d:\folder2\folder1\#;
- - ^<wrap>otherwise contents of #c:\folder1\# is moved into the
-newly created #d:\folder2\#.
+ Для папок: Якщо вказаний шлях (абсолютний чи відносний) вказує на існуючу
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:697]: "вказаний шлях вказує" is somewhat repetitive, maybe reword it somehow?

File: far/FarUkr.hlf.m4 [hunk 12]
```diff
@@ -696,14 +694,14 @@ Alt+F9 перемикає екран з 25 ліній на 50 і навпаки.
...
+папку, цільова папка буде переміщена до вказаної. Інакше - перенесена чи
+переіменована у новий шлях.
+ Таким чином якщо перенести #c:\folder1\# to #d:\folder2\#:
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:700]: Probably a comma needed after "Таким чином"

File: far/FarUkr.hlf.m4 [hunk 13]
```diff
@@ -744,22 +742,22 @@ Shift+Enter на папці з іменем "#..#" відкриває у Про
 
 @DeleteFile
 $ #Deleting and wiping files and folders#
- The following hotkeys are used to delete or wipe out files and folders:
+ Наступні сполучення використовуються для видалення файлів та папок:
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:745]: "сполучення" is somewhat ambiguous, "сполучення клавіш" maybe?

File: far/FarUkr.hlf.m4 [hunk 14]
```diff
@@ -744,22 +742,22 @@ Shift+Enter на папці з іменем "#..#" відкриває у Про
...
 
  #F8#
- If any files or folders are selected in the panel then the selected group will be deleted, otherwise
-the object currently under cursor will be deleted.
+ Якшо файли чи папки відмічені на панелі, вони будуть відалені. Інакше буде видален
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:748]: видалені

File: far/FarUkr.hlf.m4 [hunk 15]
```diff
@@ -2872,10 +2870,10 @@ $ #Вбудована програма перегляду#
 #Ctrl+Right# keys shift the content within the window one byte at a time
 in the corresponding direction.
 
- In the #hex# ~view mode~@ViewerMode@, #Alt+Left# and #Alt+Right# key
-combinations decrease or increase the number of bytes displayed on each
-row by one byte, respectively. #Ctrl+Alt+Left# and #Ctrl+Alt+Right# key
-combinations adjust the number of displayed bytes by 16 at a time.
+ В режимі #hex# ~view mode~@ViewerMode@, #Alt+Left# та #Alt+Right# комбінації
+зменьшують чи збільшують число байт, що відображається у кожному рядку на 1.
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:2874]: Probably a comma needed after "рядку"

File: far/FarUkr.hlf.m4 [hunk 16]
```diff
@@ -5039,15 +5037,15 @@ $ #Змінні середовища#
...
+ Загальна форма: #regexp# or /#regexp#/#options#.
 
- #Options#:
- #i# - ignore character case;
- #s# - ^<wrap>consider the whole text as one line, ‘#.#’ matches any character;
- #m# - ^<wrap>consider the whole text as multiple lines. ‘#^#’ and ‘#$#’ match the
-beginning and the end of any "inner" string;
+ #Опції#:
+ #i# - ігнорувати кейс;
+ #s# - ^<wrap>вважати весь текст однією строкою, ‘#.#’ буде позначати любий символ;
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:5046]: "любий" is about feelings, "будь-який" might be more appropriate.

File: far/FarUkr.hlf.m4 [hunk 17]
```diff
@@ -5039,15 +5037,15 @@ $ #Змінні середовища#
...
+ #m# - ^<wrap>вважати весь текст як декілька строк. ‘#^#’ та ‘#$#’ будут вважатися як
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:5047]: будуть

File: far/FarUkr.hlf.m4 [hunk 18]
```diff
@@ -5110,17 +5108,21 @@ zero-size expression.
  #{,m}#   - ^<wrap>equals to {0,m}
  #{,m}?#  - ^<wrap>equals to {0,m}?
 
- #"Greedy" and "not greedy" quantifiers#
+ #"Жадібні" та "не жадібний" квантифікатори#
+
+  Жадібний квантифікатор (Greedy) захоплює якомога більше символів і лише якщо
+подальший збіг не вдається, він "повертає" захоплені символи по одному,
+перевіряючи збіг регулярки (такий відкат досить дорогий за ресурсами).
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:5115]: "регулярки" is colloquial, "виразу" or "регулярного виразу" would be more appropriate.

File: far/FarUkr.hlf.m4 [hunk 19]
```diff
@@ -5110,17 +5108,21 @@ zero-size expression.
  #{,m}#   - ^<wrap>equals to {0,m}
  #{,m}?#  - ^<wrap>equals to {0,m}?
 
- #"Greedy" and "not greedy" quantifiers#
+ #"Жадібні" та "не жадібний" квантифікатори#
+
+  Жадібний квантифікатор (Greedy) захоплює якомога більше символів і лише якщо
```
Comment (MKadaner) [inline far/FarUkr.hlf.m4:5113]: Nit: Convention is to have one space indentation:
```suggestion
 Жадібний квантифікатор (Greedy) захоплює якомога більше символів і лише якщо
```

File: far/FarUkr.hlf.m4 [hunk 20]
```diff
@@ -5110,17 +5108,21 @@ zero-size expression.
...
- When expression "A.*Z" is matched to string
-"AZXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX", #.*# captures the whole string, and then
-rolls back character by character until it finds Z. On the opposite, if the expression
-is "A.*?Z" then Z is found at once. Not greedy quantifier is also known as
-#minimizing#, it captures as few characters as possible, and only if
-further match fails it captures more.
+  На противагу, якщо наш вираз буде "A.*?Z", то Z знайдено відразу, наступним кроком.
+Не жадібний квантифікатор також відомий як #lazy#, захоплює якомога менше символів
+і лише якщо подальший матч не вдається, він йде зліва направо, захоплючи наступний символ,
+та перевіряючи рядок на матчінг.
```
Comment (MKadaner) [inline far/FarUkr.hlf.m4:5125]: Here, English (and other languages') help has two paragraphs, while your translation has three. This causes the paragraph mismatch error:
```
Mismatch: Topics[@RegExp].Paragraphs.Count
    FarEng.hlf.m4: 21
    FarUkr.hlf.m4: 22
```
Please merge two last paragraphs or split this place in other help files.

**Review by MKadaner — CHANGES_REQUESTED** [review]
Thank you for helping translate Far to Ukranian. Please fix these minor issues.

File: far/FarUkr.hlf.m4 [hunk 21]
```diff
@@ -5110,17 +5108,21 @@ zero-size expression.
  #{,m}#   - ^<wrap>equals to {0,m}
  #{,m}?#  - ^<wrap>equals to {0,m}?
 
- #"Greedy" and "not greedy" quantifiers#
+ #"Жадібні" та "не жадібний" квантифікатори#
+
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:5112]: Please avoid adding or removing empty lines in HLFs.
Differences in paragraphs structure [won't pass](https://github.com/FarGroup/FarManager/actions/runs/5847752036/job/15854320652?pr=720) the build check.

File: far/FarUkr.hlf.m4 (hunk 20 above)
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:5125]: матч, матчінг - perhaps збіг, збігання?

File: far/FarUkr.hlf.m4 [hunk 22]
```diff
@@ -6804,21 +6806,21 @@ following histories:
 @Editor.F8CPs
 $ #far:config Editor.F8CPs#
 $ #far:config Viewer.F8CPs#
- These string parameters define code pages which are cycled through when
-#F8# key is pressed in ~Editor~@Editor@ or ~Viewer~@Viewer@.
+ Ці параметри визначають кодировки, що будуть використані при циклічному перебору коли натиснено
```
Comment (alabuzhev) [inline far/FarUkr.hlf.m4:6809]: "кодировки" -> "кодові сторінки"
"перебору" -> "переборі", and a comma after it.

**Comment by alabuzhev** [comment]
Thank you!

General points:
- Please keep the same paragraph structure in help, it is enforced by the build checks.
- Please spell check if possible.

File: far/farlang.templ.m4 [hunk 23]
```diff
@@ -2881,7 +2881,7 @@ upd:"ClearType-friendly redraw (can be slow)"
 "Limpiar rediseño (puede ser lento)"
 "Použiť &ClearType (pomalšie)"
 "Modalità ClearT&ype (può rallentare)"
-"Перемальовка з підтримкою ClearType (може бути повільною)"
+"Друк з підтримкою ClearType (може бути повільною)"
```
Comment (alabuzhev) [inline far/farlang.templ.m4:2884]: While technically correct, "друк" is predominantly associated with paper printing.

Maybe "вивід", as in "output", or just "rendering", since it is already used twice above?

File: far/FarUkr.hlf.m4 (hunk 20 above)
Comment (sfkulyk) [reply far/FarUkr.hlf.m4]: Reformatted to a single paragraph, thanks

File: far/farlang.templ.m4 [hunk 24]
```diff
@@ -3164,7 +3164,7 @@ upd:"Del removes blocks"
 "&Del remueve bloques"
 "Del vymaže bloky"
 "&Canc Elimina Blocchi"
-"Del видаляє б&локи"
+"Del видаляє виділений б&локи"
```
[the remark this answers is not in the document]
Comment (sfkulyk) [reply far/farlang.templ.m4]: Fixed!

File: far/farlang.templ.m4 [hunk 25]
```diff
@@ -3422,7 +3422,7 @@ upd:"DNS fully-qualified"
 upd:"DNS fully-qualified"
 "DNS Qualificato"
 "Повне DNS ім'я"
-"Поўнае імя DNS"
+"Поўнае імя DNS (FQDN)"
```
[the remark this answers is not in the document]
Comment (sfkulyk) [reply far/farlang.templ.m4:3425]: FQDN will not be extra here =)

File: far/FarUkr.hlf.m4 (hunk 17 above)
Comment (sfkulyk) [reply far/FarUkr.hlf.m4]: Thanks for catching this

File: far/FarUkr.hlf.m4 (hunk 21 above)
Comment (sfkulyk) [reply far/FarUkr.hlf.m4:5112]: fixed this paragraph

File: far/FarUkr.hlf.m4 (hunk 20 above)
Comment (sfkulyk) [reply far/FarUkr.hlf.m4]: fixed as suggested

File: far/FarUkr.hlf.m4 (hunk 22 above)
Comment (sfkulyk) [reply far/FarUkr.hlf.m4]: fixed as suggested

File: far/farlang.templ.m4 (hunk 23 above)
Comment (sfkulyk) [reply far/farlang.templ.m4]: yeah, it hard to realize the correct term in Ukrainian for that..
Replaced with "Рендерінг" for this case.

**Comment by sfkulyk** [comment]
Thanks for all comments, I have tried to fix everything. And sorry, didn't know that paragraph counting is important.
For the future will keep the same amount.
The initial goal of this update is to provide a native description of technical things, like "regular expression details" to avoid translate issues, so even if english version of the interface is preferred, the localized help can be used without misunderstanding in details.

**Comment by alabuzhev** [comment]
Thank you.
